### PR TITLE
Revision of xcube serve documentation

### PR DESCRIPTION
**Pull request by AliceBalfanz**
[Description of PR]
This PR changes documentation only. I have updated the description of server configuration file to also reflect DataStore configuration. 

Checklist:

* [ ] ~~Add unit tests and/or doctests in docstrings~~
* [ ] ~~Add docstrings and API docs for any new/modified user-facing classes and functions~~
* [x] New/modified features documented in `docs/source/*`
* [ ] ~~Changes documented in `CHANGES.md`~~
* [x] AppVeyor CI passes
* [ ] ~~Test coverage remains or increases (target 100%)~~

### CODE REVIEW

**Comment by TonioF** [comment]
For review, see https://xcube.readthedocs.io/en/alicja-xxx-update-server-docs/cli/xcube_serve.html

File: docs/source/cli/xcube_serve.rst [hunk 1]
```diff
@@ -471,11 +489,75 @@ Example
     [I 190924 17:08:54 service:159] press CTRL+C to stop service
 
 
+Server Demo Configuration File for DataStores
+=============================================
+The server configuration file consists of various parts, some of them are necessary others are optional.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:494]:
```suggestion
The server configuration file consists of various parts, some of them are necessary, others are optional.
```

File: docs/source/cli/xcube_serve.rst [hunk 2]
```diff
@@ -471,11 +489,75 @@ Example
...
+Here the `demo_stores configuration file`_ used in the :ref:`example-stores` is explained in detail.
+
+This configuration file differs only in one part compared to :ref:`..config`. This part is :ref:`data-stores`,
+the other main parts :ref:`authentication`, :ref:`dataset-attribution`,
+:ref:`place-groups` and :ref:`styles`. Can be used in combination with :ref:`data-stores`.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:499]:
```suggestion
This configuration file differs only in one part compared from :ref:`..config`. This part is :ref:`data-stores`.
The other main parts (:ref:`authentication`, :ref:`dataset-attribution`,
:ref:`place-groups`, and :ref:`styles`) can be used in combination with :ref:`data-stores`.
```

File: docs/source/cli/xcube_serve.rst [hunk 3]
```diff
@@ -471,11 +489,75 @@ Example
...
+*root* [mandatory] gives the common part of the path of the served datasets.
+*max_depth* [optional] if wildcard is used in *Dataset Path* this indicated, how far the server should step down and
+serve the discovered datasets.
+*storage_options* [optional] is necessary when serving datasets from the cloud. With *anon* the accessibility is
+configured, if the datasets are public-read, *anon* is set to "true", "false" indicates they are protected.
+
+*Datasets* [optional] if not specified, every dataset in the indicated location supported by xcube will be read and
+served by xcube serve. In order to filter certain datasets you can list Paths that shall be served by xcube serve.
+*Path* may contain wildcards. Each Dataset entry may have *Styles* and *PlaceGroups* associated with them, the same way
+as in :ref:`..config`.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:539]: Have a look at the layout on rtd

File: docs/source/cli/xcube_serve.rst [hunk 4]
```diff
@@ -471,11 +489,75 @@ Example
...
+          - Path: "*2.zarr"
+            Style: "default"
+            # ChunkCacheSize: 1G
+
+*Identifier* [mandatory] is a unique ID for each DataStore.
+
+*StoreID* [mandatory] can be *file* for locally stored datasets and *s3* for datasets located in the cloud.
+
+*StoreParams* [mandatory]
+*root* [mandatory] gives the common part of the path of the served datasets.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:530]:
```suggestion
*root* [mandatory] states a common beginning part of the paths of the served datasets.
```

File: docs/source/cli/xcube_serve.rst [hunk 5]
```diff
@@ -471,11 +489,75 @@ Example
...
+*max_depth* [optional] if wildcard is used in *Dataset Path* this indicated, how far the server should step down and
+serve the discovered datasets.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:532]:
```suggestion
*max_depth* [optional] if wildcard is used in *Dataset Path* this indicates how far the server should step down in the directory structure to detect datasets that shall be served.
```

File: docs/source/cli/xcube_serve.rst [hunk 6]
```diff
@@ -471,11 +489,75 @@ Example
...
+*storage_options* [optional] is necessary when serving datasets from the cloud. With *anon* the accessibility is
+configured, if the datasets are public-read, *anon* is set to "true", "false" indicates they are protected.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:534]: Add "Credentials" my be set with keywords 'key' and 'secret'."

**Review by TonioF — CHANGES_REQUESTED** [review]
See comments. Also, the references in section "Server Demo Configuration File for DataStores" do not work.

File: docs/source/cli/xcube_serve.rst [hunk 7]
```diff
@@ -34,65 +35,62 @@ Serve data cubes via web service.
...
       -s, --show               Run viewer app. Requires setting the environment
                                variable XCUBE_VIEWER_PATH to a valid xcube-viewer
                                deployment or build directory. Refer to
                                https://github.com/dcs4cop/xcube-viewer for more
                                information.
-
-      -v, --verbose            Delegate logging to the console (stderr).
-      --traceperf              Print performance diagnostics (stdout).
+      -q, --quiet              Disable output of log messages to the console
+                               entirely. Note, this will also suppress error and
```
[the remark this answers is not in the document]
Comment (AliceBalfanz) [reply docs/source/cli/xcube_serve.rst:79]: This is the output of xcube serve --help, I will not change it here.

File: docs/source/cli/xcube_serve.rst [hunk 8]
```diff
@@ -34,65 +35,62 @@ Serve data cubes via web service.
...
+                               warning messages.
+      -v, --verbose            Enable output of log messages to the console. Has
+                               no effect if --quiet/-q is used. May be given
+                               multiple times to control the level of log
+                               messages, i.e., -v refers to level INFO, -vv to
```
[the remark this answers is not in the document]
Comment (AliceBalfanz) [reply docs/source/cli/xcube_serve.rst:84]: This is the output of xcube serve --help, I will not change it here.

File: docs/source/cli/xcube_serve.rst [hunk 9]
```diff
@@ -221,29 +230,38 @@ Remotely Stored xcube Datasets
...
 The above example of how to specify a xcube dataset to be served above is using a datacube stored in
-an S3 bucket within the OpenTelekomCloud.
+an S3 bucket within the Amazon Cloud. Please have a closer look to the parameter *Anonymous: true*.
+This means, the datasets permissions are set to public read. If you have a dataset that is not public-public read, set
+*Anonymous: false*. Furthermore, you need to have valid credentials on the machine where the server runs.
+Credentials may be saved either in a file called .aws/credentials with content like below:
+
+    [default]
+    aws_access_key_id=AKIAIOSFODNN7EXAMPLE
+    aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
```
[the remark this answers is not in the document]
Comment (AliceBalfanz) [reply docs/source/cli/xcube_serve.rst:240]: They are :)

File: docs/source/cli/xcube_serve.rst [hunk 10]
```diff
@@ -143,42 +140,52 @@ and the it is referred to by the non-hidden, actual dataset using the ``TimeSeri
...
 
       - ...
 
 
+.. _config:
 Server Demo Configuration File
 ==============================
 The server configuration file consists of various parts, some of them are necessary others are optional.
-Here the `demo configuration file`_ used in the :ref:`example` is explained in detail.
+Here the `demo configuration file`_ used in the `example`_ is explained in detail.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:160]: The example link does not work as you have two examples in this file. Can you rename them appropriately?

File: docs/source/cli/xcube_serve.rst [hunk 11]
```diff
@@ -143,42 +140,52 @@ and the it is referred to by the non-hidden, actual dataset using the ``TimeSeri
...
 
-The configuration file consists of five main parts :ref:`authentication`, :ref:`dataset-attribution`, :ref:`datasets`,
-:ref:`place-groups` and :ref:`styles`.
+The configuration file consists of five main parts `authentication`_, `dataset attribution`_, `datasets`_,
+`place groups`_ and `styles`_.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:163]:
```suggestion
The configuration file consists of five main parts `authentication-optional`_, `dataset attribution-optional`_, `datasets-mandatory`_,
`place groups-optional`_ and `styles-optional`_.
```

File: docs/source/cli/xcube_serve.rst [hunk 12]
```diff
@@ -221,41 +230,60 @@ Remotely Stored xcube Datasets
...
+
+    | [default]
+    | aws_access_key_id=AKIAIOSFODNN7EXAMPLE
+    | aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+
+Or they may be exported as environment variables AWS_SECRET_ACCESS_KEY and AWS_ACCESS_KEY_ID.
 
-*Identifier* [mandatory] is a unique ID for each xcube dataset, it is ment for machine-to-machine interaction
+Further down an example for a `locally-stored-xcube-datasets`_ will be given,
+as well as an example of a `on-the-fly-generation-of-xcube-datasets`_.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:245]: Links are not working.

File: docs/source/cli/xcube_serve.rst [hunk 13]
```diff
@@ -221,41 +230,60 @@ Remotely Stored xcube Datasets
...
-in the server configuration file. The usage of *Styles* is described in section :ref:`styles`.
+*Styles* [optional]
+influence the visualization of the xucbe dataset in the xcube viewer if specified in the server configuration file. The usage of *Styles* is described in section `styles`_.
 
-*PlaceGroups* [optional] allow to associate places (e.g. polygons or point-location) with a particular xcube dataset.
+*PlaceGroups* [optional]
+allow to associate places (e.g. polygons or point-location) with a particular xcube dataset.
 Several different place groups may be connected to a xcube dataset, these different place groups are distinguished by
-the *PlaceGroupRef*. The configuration of *PlaceGroups* is described in section :ref:`place-groups`.
+the *PlaceGroupRef*. The configuration of *PlaceGroups* is described in section `place groups`_.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:280]:
```suggestion
the *PlaceGroupRef*. The configuration of *PlaceGroups* is described in section `place-groups-optional`_.
```

File: docs/source/cli/xcube_serve.rst [hunk 14]
```diff
@@ -221,41 +230,60 @@ Remotely Stored xcube Datasets
...
 
-*AccessControl* [optional] can only be used when providing :ref:`authentication`. Datasets may be protected by
+*AccessControl* [optional]
+can only be used when providing `authentication`_. Datasets may be protected by
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:283]:
```suggestion
can only be used when providing `authentication-optional`_. Datasets may be protected by
```

**Review by TonioF — CHANGES_REQUESTED** [review]
Almost none of the document-internal links work. Did you check them in the generated docs?

File: docs/source/cli/xcube_serve.rst [hunk 15]
```diff
@@ -471,11 +521,78 @@ Example
     [I 190924 17:08:54 service:159] press CTRL+C to stop service
 
 
+Server Demo Configuration File for DataStores
+=============================================
+The server configuration file consists of various parts, some of them are necessary, others are optional.
+Here the `demo stores configuration file`_ used in the `example stores`_ is explained in detail.
+
+This configuration file differs only in one part compared from `config`_. This part is `data stores`_.
```
Comment (TonioF) [inline docs/source/cli/xcube_serve.rst:529]:
```suggestion
This configuration file differs only in one part from :ref:`config`.
`the configuration above`_: `data stores`_.
```

Most of the configuration of locally stored datasets is equal to the configuration of
:ref:`remotely-stored-xcube-datasets`.
`remotely stored xcube datasets`_.

**Review by TonioF — APPROVED** [review]
One suggestion for improvement. However, I was not able to test whether this really works the way I want it, and I am okay if you don't accept it.

**Pull request merged**